### ISSUE TITLE
Does iflink ever worked?

### ISSUE THREAD

**Issue by MedourMehdi** [freemint/freemint#363]
Hi,

Obviously the _[iflink](https://github.com/freemint/freemint/blob/master/tools/net-tools/iflink.c#L86)_ binary provided can't work.
It calls _[SIOCGIFNAME](https://github.com/freemint/freemint/blob/master/tools/net-tools/iflink.c#L86)_ who do nothing except check for permissions and then lead to an error.
It does bad usage of _[SIOCGIFNAME](https://github.com/freemint/freemint/blob/master/tools/net-tools/iflink.c#L61)_ with a wrong iflink structure and then lead to an error.

Does anyone know it ever worked? May be I misunderstand what is it for.

**Comment by th-otto** [comment]
What make you think so? The call should end up in https://github.com/freemint/freemint/blob/0c8bc27e427d5b2097b5db07d185ca5cdffa44e2/sys/sockets/inet4/if.c#L590 , which then calls the ioctl of the driver to get the devicename. Only problem seems to be that `SIOCGIFNAME` does not seem to be implemented in most drivers, except for ppp (https://github.com/freemint/freemint/blob/0c8bc27e427d5b2097b5db07d185ca5cdffa44e2/sys/sockets/xif/ppp.c#L324) and slip (https://github.com/freemint/freemint/blob/0c8bc27e427d5b2097b5db07d185ca5cdffa44e2/sys/sockets/xif/slip.c#L217)

**Comment by MedourMehdi** [comment]
Agreed but this isn't the role of SIOCGIFNAME as It should return an index.
Do we have to deal with sizeof(arg) to know if we should call the ioctl of the driver?

**Comment by MedourMehdi** [comment]
What do you think about filtering with

if(( nif->hwtype & HWTYPE_PPP ) || ( nif->hwtype & HWTYPE_SLIP )){

}

I don't know how to deal with that.

**Comment by th-otto** [comment]
Most simply, use a new define to map an index to interface name. That means also changes to your PR in mintlib.

**Comment by MedourMehdi** [comment]
I've made some changes using another define (SIOCGIFNAME_ETH), I can change the name if you want.

**Comment by MedourMehdi** [comment]
> Most simply, use a new define to map an index to interface name. That means also changes to your PR in mintlib.
Hi,

What do you think about using this in mintlib sockios.h ?
`#ifdef __USE_GNU

#ifdef SIOCGIFNAME
#undef SIOCGIFNAME
#define SIOCGIFNAME SIOCGIFNAME_ETH
#endif

#endif
`
Whatever what's blocking the PR? It seems all good to me.
